### DISCUSSION TITLE
ImportError with Gradio

### DISCUSSION THREAD

**Discussion by MiguelHGDC** [mrdbourke/pytorch-deep-learning#640] [Q&A] [answered]
```
try:
    import gradio as gr
except: 
    !pip -q install gradio
    import gradio as gr
    
print(f"Gradio version: {gr.__version__}")
```
Output: **ImportError**: cannot import name 'deprecated' from 'typing_extensions'

My versions:

python version: 3.8.10
torch: '2.0.1+cu118'
torchvision: '0.15.2+cu118'
typing_extensions: 4.7.1
gradio==3.44.3

Does anyone know what the compatibility problem is?

**Comment by MariaKrakowiak** [comment]
Hi @MiguelHGDC ,

Did you manage to solve above issue? I have similar one.

**Comment by mrdbourke** [comment, marked as answer]
Hi @MiguelHGDC and @MariaKrakowiak ,

Are you still able to use Gradio? 

Are you using Google Colab or a local machine?

I just tried to install/import it in Google Colab and it's showing a similar error but the conflict is with `tensorflow-probability` (not needed for the course or to use Gradio.

See the notebook here: https://colab.research.google.com/drive/1f2-ptPGecjPQ0zMNnS6rTVyHmR8i6izv?usp=sharing 

<img width="1687" alt="Screen Shot 2023-10-26 at 4 00 19 pm" src="https://github.com/mrdbourke/pytorch-deep-learning/assets/16750345/f33c7f16-2adc-41fa-9300-106978d168e2">

However, this conflict does not influence the use of Gradio.

**Reply by mrdbourke** [reply]
Also, perhaps this thread will help: https://github.com/mrdbourke/pytorch-deep-learning/discussions/699 (similar issue)

**Reply by MariaKrakowiak** [reply]
Hi @mrdbourke ,

Thanks for your advice. In completely new Colab it works but unfortunately in my case it doesn't.

The thing is, I am unable to use Gradio. Before installing it, I have checked the typing_extensions version (!pip show typing_extensions) which is problematic and I received 4.5.0.

Then, I have installed Gradio via !pip install gradio with no errors and the typing_extensions version showed 4.8.0.

Finally, I tried to import the module but I have received:

![image](https://github.com/mrdbourke/pytorch-deep-learning/assets/59764846/a9354f04-ac3d-4e09-8045-d7e63785799b)

In Github dedicated for this extension (https://github.com/python/typing_extensions/blob/main/CHANGELOG.md) there is information that Doc is available from version 4.8.0 which I have already installed.

Do you have any idea how to proceed? 

I tried restarting and reinstalling runtime few times with no result. I am running out of ideas.

**Reply by MiguelHGDC** [reply]
I remember trying to fix this error, and I encountered the same issue as you, @MariaKrakowiak. I couldn't find the correct versions; I'm sorry I didn't respond.

Locally, it's true that it's not working... @mrdbourke,  perhaps the correct solution would be to create different virtual environments for each task, especially one specifically for using Gradio.

**Reply by MariaKrakowiak** [reply]
Hi @MiguelHGDC ,

No problem - I finally overcame the issue.

My solution was to import the gradio at the beginning of Colab notebook by:

```
!pip install gradio
import gradio as gr
```

Thanks to this there were no issues with other extensions versions.

Sometimes the simplest solutions are the best 👍
